### DISCUSSION TITLE
Thanos storegateway "exceeded chunks limit"

### DISCUSSION THREAD

**Discussion by tieungao88** [thanos-io/thanos#5956] [Questions & Answers]
My thanos storegateway have problem, It query data from S3.

`level=debug ts=2022-12-08T23:29:24.811544401Z caller=bucket.go:1141 msg="stats query processed" request="min_time:1669244100000 max_time:1670540400000 matchers:<type:NEQ name:\"cluster\" > matchers:<type:RE name:\"workload\" value:\"face-matching-service|my-keycloak|user-profile-service|ocr-service|event-service|resources-service|m4c-notification-service|loan-service|authentication-service\" > matchers:<name:\"__name__\" value:\"namespace_workload_pod:kube_pod_owner:relabel\" > aggregates:COUNT aggregates:SUM " stats="&{blocksQueried:0 postingsTouched:29 PostingsTouchedSizeSum:7KiB542B postingsToFetch:0 postingsFetched:0 PostingsFetchedSizeSum:0B postingsFetchCount:0 PostingsFetchDurationSum:0s cachedPostingsCompressions:0 cachedPostingsCompressionErrors:0 CachedPostingsOriginalSizeSum:0B CachedPostingsCompressedSizeSum:0B CachedPostingsCompressionTimeSum:0s cachedPostingsDecompressions:29 cachedPostingsDecompressionErrors:0 CachedPostingsDecompressionTimeSum:132.063µs seriesTouched:0 SeriesTouchedSizeSum:0B seriesFetched:0 SeriesFetchedSizeSum:0B seriesFetchCount:0 SeriesFetchDurationSum:0s chunksTouched:0 ChunksTouchedSizeSum:0B chunksFetched:0 ChunksFetchedSizeSum:0B chunksFetchCount:0 ChunksFetchDurationSum:0s GetAllDuration:0s mergedSeriesCount:0 mergedChunksCount:0 MergeDuration:0s}" err="rpc error: code = Aborted desc = fetch series for block 01GKSZF5KPFGESADDCP2XHVW4B: exceeded chunks limit: limit 4 violated (got 6)"`

This is my config:
  extraFlags:
  - --store.grpc.series-max-concurrency=2
  - --store.enable-index-header-lazy-reader
  - --index-cache-size=500MB
  - --store.grpc.series-sample-limit=500
  - --chunk-pool-size=10GB
I'm using: https://artifacthub.io/packages/helm/bitnami/thanos version 11.6.4

Please help me !
Thanks.

**Comment by yeya24** [comment]
So you configured `store.grpc.series-sample-limit` value as 500 and the maximum chunks limit is calculated via `store.grpc.series-sample-limit/120` which is 4 in your case. From the error message you are trying to query 6 chunks so it exceeds the limit.

Please consider to increase the value of `store.grpc.series-sample-limit` or set it to 0 so you get no limit on chunks.

Hope this helps! Since it is not a bug, I will convert it to a discussion instead.
